disnake/state.py

```diff
@@ -912,11 +922,11 @@ def parse_interaction_create(self, data) -> None:
             elif interaction.data.component_type is ComponentType.select:
                 self.dispatch("dropdown", interaction)
 
-        elif interaction_type == 4:
+        elif data["type"] == 4:
             interaction = ApplicationCommandInteraction(data=data, state=self)
             self.dispatch("application_command_autocomplete", interaction)
 
-        elif interaction_type == 5:
+        elif data["type"] == 5:
```

Comment:
Could we please revert these changes?

Reply:
Doing it this way, without an intermediate variable, helps pyright narrow the data union to the correct type - otherwise we'd have to `# type: ignore` all the `*Interaction(data=data, state=self)` calls, which I'd really like to avoid.
The performance overhead is negligible, with a difference of `(10.9 us - 8.65 us) / 100 = 22.5 ns` per call: https://workbin.dev/?id=1658072299515132

Reply:
That's unfortunate, I suppose. Should probably have a comment explaining why

Reply:
https://github.com/DisnakeDev/disnake/pull/594/commits/fc2be126a25524eda99b3018e8124c48d2574f31